src/cryptography/x509/base.py

```diff
@@ -438,10 +453,28 @@ def load_der_x509_crl(
 
 
 class CertificateSigningRequestBuilder(object):
-    def __init__(self, subject_name=None, extensions=[], attributes=[]):
+    def __init__(
+        self,
+        subject_name: typing.Optional[Name] = None,
+        extensions: typing.Optional[
+            typing.Iterable[Extension[ExtensionType]]
+        ] = None,
+        attributes: typing.Optional[
+            typing.Iterable[typing.Tuple[ObjectIdentifier, bytes]]
+        ] = None,
+    ):
         """
         Creates an empty X.509 certificate request (v1).
         """
+        if extensions is None:
+            extensions = []
+        else:
+            extensions = list(extensions)
+        if attributes is None:
+            attributes = []
+        else:
+            attributes = list(attributes)
```

Comment:
We don't need the `list` wrapping here -- this function isn't a public API and our internals only ever call it with a `list`. For the same reason it's fine to just use the `[]` default.

Reply:
The class is documented right here: https://cryptography.io/en/latest/x509/reference.html. While constructor arguments are admittedly not documented, it's not hard to find out about them, and the names are not prefixed with a `_` either. So I really feel we should leave this in. It's not like it hurts anyone.